event/event.go

```diff
@@ -5,12 +5,38 @@ import (
 	"time"
 )
 
+// Category define how to display the Category.
+type Category int32
+
+const (
```

Comment:
@mdubbyap gomeatalinter complained about "USER_DEFINED" and "SERVICE_DISCOVERY" because there are underscores in the name which is anti-go.  It actually failed circle ci for that reason.  So writing this as "USERDEFINED" was the workaround.